pages/assessment.tsx

```diff
@@ -15,13 +15,33 @@ import { Button } from "@mui/material";
 import Header from "../components/header";
 import findIndex from "lodash/findIndex";
 import cloneDeep from "lodash/cloneDeep";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
 
 const Assessment: NextPage = () => {
-  const taskNumber: number = 2; // Later will get it another way. And then make sure tasknumber doesn't get too high.
-  const taskTitle: string =
-    data.ext_inspera_candidates[0].result.ext_inspera_questions[taskNumber - 1]
-      .ext_inspera_questionTitle;
+  // create router object
+  const router = useRouter();
 
+  // isReady: boolean - checks whether the router fields are updated client-side and ready for use.
+  useEffect(() => {
+    if (!router.isReady) return;
+    setTaskNumber(router.query.task);
+    setTaskTitle(
+      data.ext_inspera_candidates[0].result.ext_inspera_questions[
+        router.query.task - 1
```

Comment:
This breaks the page if you insert a wrong number in the URL. But we are aware, so we should just make it a separate issue.

Reply:
There should be something like if router.query.task == undefined --> send to error page. There is an issue on it #54

Reply:
Nicee!